Framework/Algorithms/src/ChangePulsetime.cpp

```diff
@@ -19,16 +20,13 @@ using std::size_t;
 /** Initialize the algorithm's properties.
  */
 void ChangePulsetime::init() {
-  declareProperty(make_unique<WorkspaceProperty<EventWorkspace>>(
-                      "InputWorkspace", "", Direction::Input),
+  declareProperty(createWithWorkspaceIndices<EventWorkspace>("InputWorkspace"),
```

Comment:
This is still a bit cumbersome to type. Would it possibly make sense to merge the helper function with the `declareProperty`, such that we can do something like this:

```cpp
declareWorkspacePropertyWithIndex<EventWorkspace>("InputWorkspace");
// or
declarePropertyWithIndex<EventWorkspace>("InputWorkspace");
```
?